### DISCUSSION TITLE
React course: is memoization still relevant?

### DISCUSSION THREAD

**Discussion by Asmatzaile** [TheOdinProject/curriculum#29207] [General]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

If I understood it correctly, React 19 will automatically memoize most expensive calculations, like the one described in the screenshot below:
![irudia](https://github.com/user-attachments/assets/346209c6-547c-48f9-ae56-2794a0f5cd62)
I still don't know enough to determine whether the information is accurate, but I thought to point it out for you to look at :)

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-react-new-refs-and-memoization

**Comment by wise-king-sullyman** [comment]
I haven't tried 19 yet to confirm, but my understanding is that the React compiler (which will handle performance aspects such as memoization automatically) is a separate thing from React 19, that the compiler is still in beta at this time, and that it will be an opt in thing that requires modifying the build step of projects it is used in.

Either way I think covering manual memoization will still be relevant for quite some time. I know of projects at my work that are still using 16.8.

Once the compiler is out as stable it is probably something we should talk about though.

Tagging @TheOdinProject/react for awareness/thoughts and to correct me if yall know anything I don't.

**Comment by 01zulfi** [comment]
^ seconded

The "automatic memoization" is a React Compiler feature which is separate from React 19. The following article is an informative read https://www.developerway.com/posts/react-compiler-soon

We can open this convo after the compiler is stable.

**Comment by mao-sz** [comment]
Reviving this because React Compiler is stable now and we've added a section on it in the [Refs and Memoization lesson](https://www.theodinproject.com/lessons/node-path-react-new-refs-and-memoization#react-compiler).

Just because it's stable and does auto-memo for most cases, doesn't make learning about memoization irrelevant. If you don't know what it is, you won't know what RC is doing for you, and the world won't suddenly switch every codebase to use RC and rid themselves of manual memo straightaway. Same as why we still have content on class-based components. They're not "irrelevant" in the real world. Also, even React docs say there may be cases where you need more control with memoisation than RC can automatically handle for you.

So I believe this discussion can be closed now.
